hoomd/data/parameterdicts.py

```diff
@@ -345,6 +346,13 @@ def __len__(self):
         """int: The number of keys."""
         return len(self._dict)
 
+    def __eq__(self, other):
+        """Equality between ParameterDict objects."""
+        if not isinstance(other, ParameterDict):
+            return NotImplemented
+        return (set(self.keys()) == set(other.keys()) and np.all(
+            np.all(self[key] == other[key]) for key in self.keys()))
```

Comment:
I have the same question on this PR as in #965. Is it valid to use numpy for this check on keys that may not be numpy arrays?

Reply:
Yes, this should be valid. When passed a Boolean returns that Boolean value, and when passed a sequence computes the n-wise conjunction as expected.